### PR TITLE
[Term Entry] Go Math Functions Exp2()

### LINKED ISSUE #2828: [Term Entry] Go Math Functions Exp2()
### Reason/inspiration (optional)

We would like a new term entry for [Go](https://www.codecademy.com/resources/docs/go) about the `Exp2()` function.

The entry should go in a new file in `docs/content/go/concepts/math-functions/terms/exp2/exp2.md`

The entry should 

- Have a syntax section illustrating the function with a ` ```pseudo ` code block. 
- An example showing the function in use with a ` ```go ` code block. 
- A runnable codebyte example in a ` ```codebyte/golang` block. 

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Manikanta528**
### Description

closes #2828 

### Type of Change

- Adding a new entry Go Math Functions Exp2()

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/math-functions/terms/exp2/exp2.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import (
+  "fmt"
+  "math"
+)
+
+func main() {
+    fmt.Println(math.Exp2(0))
+	fmt.Println(math.Exp2(math.Inf(1)))
+	fmt.Println(math.Exp2(math.NaN()))
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/exp2/exp2.md:75]:
```suggestion
  fmt.Println(math.Exp2(0))
  fmt.Println(math.Exp2(math.Inf(1)))
  fmt.Println(math.Exp2(math.NaN()))
```

File: content/go/concepts/math-functions/terms/exp2/exp2.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+```
+
+Where `result` will contain the value of base-2 exponential of `x`, returned as type `float64`.
+Special cases:
+
+- The result of `Exp2(+Inf)` is +Inf
+- The result of `Exp2(NaN)` is NaN
+  
+## Example
+
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/exp2/exp2.md:33]: There should be a brief introductory sentence before the example

File: content/go/concepts/math-functions/terms/exp2/exp2.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
+---
+Title: 'Exp2()'
+Description: 'Returns the value of base-2 exponential of given number.'
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/exp2/exp2.md:3]:
```suggestion
Description: 'Returns the value of a base-2 exponential of a given number.'
```

File: content/go/concepts/math-functions/terms/exp2/exp2.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+Tags:
+  - 'Math'
+  - 'Arithmetic'
+  - 'Functions'
+CatalogContent:
+  - 'learn-go'
+  - 'paths/computer-science'
+---
+
+The **`Exp2()`** function returns the value of base-2 exponential of given number of type `float64`.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/exp2/exp2.md:16]:
```suggestion
The **`Exp2()`** function returns the value of a base-2 exponential of a given number of type `float64`.
```

File: content/go/concepts/math-functions/terms/exp2/exp2.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+## Syntax
+
+```pseudo
+import math
+
+result := math.Exp2(x)
+```
+
+Where `result` will contain the value of base-2 exponential of `x`, returned as type `float64`.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/exp2/exp2.md:26]:
```suggestion
Where `result` will contain the value of a base-2 exponential of `x`, returned as a `float64`.
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@Manikanta528 I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

File: content/go/concepts/math-functions/terms/exp2/exp2.md (hunk 2 above)
Comment (Manikanta528) [reply content/go/concepts/math-functions/terms/exp2/exp2.md:33]: For example?

Comment (caupolicandiaz) [reply content/go/concepts/math-functions/terms/exp2/exp2.md:33]: This is typical in our entries. Please review other docs if you need a reference.

Comment (Manikanta528) [reply content/go/concepts/math-functions/terms/exp2/exp2.md:33]: I am doing it with reference to other similar docs

**Review by caupolicandiaz — APPROVED** [review]
@Manikanta528 I'm forwarding this along for a second review.

**Review by SSwiniarski — APPROVED** [review]
@Manikanta528 Looks good.

**Comment by SSwiniarski** [comment]
@Manikanta528 Congrats on the new entry! Live link here: https://www.codecademy.com/resources/docs/go/math-functions/exp2

image

**Pull request merged**